src/libtls/tls_peer.c

```diff
@@ -1352,35 +1358,21 @@ static status_t send_client_hello(private_tls_peer_t *this,
 	extensions->write_data16(extensions, signatures->get_buf(signatures));
 	signatures->destroy(signatures);
 
-	if (this->dh)
+
+	DBG2(DBG_TLS, "sending extension: %N",
+		 tls_extension_names, TLS_EXT_KEY_SHARE);
+	extensions->write_uint16(extensions, TLS_EXT_KEY_SHARE);
+	ext_content = bio_writer_create(32);
+	if (!tls_write_key_share(ext_content, selected_curve, this->dh))
```

Comment:
To avoid having to allocate the writer blindly (by the way, 32 is the default if 0 is passed), it would also be possible to pass a pointer to the writer to the function so it can allocate it with `pub.len + 7` (`6` didn't leave room for the legacy encoding). It would then also not have to be destroyed in the error case.